src/Utils/Rotation.cpp

```diff
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
 #ifndef SKIP_ASSERT
     double sqrtNorm = static_cast<double>(this->squaredNorm());
     bool checkUnit(fabs(sqrtNorm - 3.) < 1e-4);
-    if (!checkUnit){
-        // Make calulation here
-        utils::Matrix3d mat;
+    if (!checkUnit)
```

Comment:
This looks good overall, I fell now that we should move this code into the proper class (I can help you with that) and write tests on this (I can help you with that too :) )

Thanks for the amazing work again!

Reply:
Indeed, I would appreciate if you could help me with that :)